src/test/java/org/betonquest/betonquest/api/bukkit/config/custom/multi/MultiSectionConfigurationModificationTest.java

```diff
@@ -61,13 +60,73 @@ public class MultiSectionConfigurationModificationTest {
-        config3 = YamlConfiguration.loadConfiguration(new File("src/test/resources/api/bukkit/multi/modification/config3.yml"));
+        configAll = new YamlConfiguration();
+        setupConfig1(configAll);
+        setupConfig2(configAll);
+        setupConfig3(configAll);
+        config1 = setupConfig1(new YamlConfiguration());
+        config2 = setupConfig2(new YamlConfiguration());
+        config3 = setupConfig3(new YamlConfiguration());
         config = getConfig();
     }
 
+    private Configuration setupConfig1(final Configuration config) {
+        final ConfigurationSection pathSection = config.createSection("events.custom.path");
+        pathSection.set("give.type", "give");
+        pathSection.set("give.item", "emerald");
+        pathSection.set("give.amount", 5);
+        pathSection.set("take.type", "give");
+        pathSection.set("take.item", "emerald");
+        pathSection.set("take.amount", 5);
+
+        final ConfigurationSection journalSection = config.createSection("events.custom.journal");
+        journalSection.set("type", "journal");
+        journalSection.set("action", "add");
+        journalSection.set("target", "quest_started");
+
+        return config;
+    }
+
+    private Configuration setupConfig2(final Configuration config) {
+        final ConfigurationSection pathSection = config.createSection("events.custom.path");
+        pathSection.set("notify.type", "notify");
+        pathSection.set("notify.message", "Example Message");
+        pathSection.set("notify.io", "chat");
+        pathSection.set("cancel.type", "cancel");
+        pathSection.set("cancel.target", "myQuest");
+
+        final ConfigurationSection emeraldSection = config.createSection("conditions.items.emerald");
+        emeraldSection.set("type", "item");
+        emeraldSection.set("item", "emerald");
+        emeraldSection.set("amount", 5);
+
+        return config;
+    }
+
+    private Configuration setupConfig3(final Configuration config) {
+        final ConfigurationSection finishedSection = config.createSection("events.custom.finished");
+        finishedSection.set("type", "notify");
+        finishedSection.set("message", "Quest Finished");
+        finishedSection.set("io", "advancement");
+
+        final ConfigurationSection startSection = config.createSection("events.start");
+        startSection.set("type", "notify");
+        startSection.set("message", "Quest Started");
+        startSection.set("io", "advancement");
+
+        final ConfigurationSection stoneSection = config.createSection("conditions.items.stone");
+        stoneSection.set("type", "item");
+        stoneSection.set("item", "stone");
+        stoneSection.set("amount", 3);
+
+        final ConfigurationSection itemIronSection = config.createSection("conditions.item_iron");
+        itemIronSection.set("type", "item");
+        itemIronSection.set("item", "iron");
+        itemIronSection.set("amount", 2);
+
+        return config;
+    }
```

Comment:
Repeatedly calling `createSection` will override previously created sections.

I'd suggest using a method like this to make sure that the section is only created if it didn't already exist:
```java
    private static ConfigurationSection getOrCreateSection(final Configuration config, final String path) {
        return Objects.requireNonNullElseGet(config.getConfigurationSection(path), () -> config.createSection(path));
    }
```

Such a change would look like this:
```suggestion
    private Configuration setupConfig1(final Configuration config) {
        final ConfigurationSection pathSection = getOrCreateSection(config, "events.custom.path");
        pathSection.set("give.type", "give");
        pathSection.set("give.item", "emerald");
        pathSection.set("give.amount", 5);
        pathSection.set("take.type", "give");
        pathSection.set("take.item", "emerald");
        pathSection.set("take.amount", 5);

        final ConfigurationSection journalSection = getOrCreateSection(config, "events.custom.journal");
        journalSection.set("type", "journal");
        journalSection.set("action", "add");
        journalSection.set("target", "quest_started");

        return config;
    }

    private Configuration setupConfig2(final Configuration config) {
        final ConfigurationSection pathSection = getOrCreateSection(config, "events.custom.path");
        pathSection.set("notify.type", "notify");
        pathSection.set("notify.message", "Example Message");
        pathSection.set("notify.io", "chat");
        pathSection.set("cancel.type", "cancel");
        pathSection.set("cancel.target", "myQuest");

        final ConfigurationSection emeraldSection = getOrCreateSection(config, "conditions.items.emerald");
        emeraldSection.set("type", "item");
        emeraldSection.set("item", "emerald");
        emeraldSection.set("amount", 5);

        return config;
    }

    private Configuration setupConfig3(final Configuration config) {
        final ConfigurationSection finishedSection = getOrCreateSection(config, "events.custom.finished");
        finishedSection.set("type", "notify");
        finishedSection.set("message", "Quest Finished");
        finishedSection.set("io", "advancement");

        final ConfigurationSection startSection = getOrCreateSection(config, "events.start");
        startSection.set("type", "notify");
        startSection.set("message", "Quest Started");
        startSection.set("io", "advancement");

        final ConfigurationSection stoneSection = getOrCreateSection(config, "conditions.items.stone");
        stoneSection.set("type", "item");
        stoneSection.set("item", "stone");
        stoneSection.set("amount", 3);

        final ConfigurationSection itemIronSection = getOrCreateSection(config, "conditions.item_iron");
        itemIronSection.set("type", "item");
        itemIronSection.set("item", "iron");
        itemIronSection.set("amount", 2);

        return config;
    }

    private static ConfigurationSection getOrCreateSection(final Configuration config, final String path) {
        return Objects.requireNonNullElseGet(config.getConfigurationSection(path), () -> config.createSection(path));
    }
```